lib/node_modules/@stdlib/blas/scopy/test/test.js

```diff
@@ -0,0 +1,693 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Float32Array = require( '@stdlib/array/float32' );
+var Int16Array = require( '@stdlib/array/int16' );
+var array = require( '@stdlib/ndarray/array' );
+var zeros = require( '@stdlib/ndarray/zeros' );
+var ndarray = require( '@stdlib/ndarray/ctor' );
+var scopy = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof scopy, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 2', function test( t ) {
+	t.strictEqual( scopy.length, 2, 'has expected arity' );
+	t.end();
+});
+
+tape( 'the function throws an error if provided a first argument which is not a non-zero-dimensional ndarray containing single-precision floating-point numbers', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		'5',
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		function noop() {},
+		zeros( [], {
+			'dtype': 'float32'
+		}),
+		array( new Int16Array( 10 ) )
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		var y = array( new Float32Array( 10 ) );
+
+		return function badValue() {
+			scopy( value, y );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a first argument which is not a non-zero-dimensional ndarray containing single-precision floating-point numbers (dimension)', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		'5',
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		function noop() {},
+		zeros( [], {
+			'dtype': 'float32'
+		}),
+		array( new Int16Array( 10 ) )
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided ' + values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		var y = array( new Float32Array( 10 ) );
+
+		return function badValue() {
+			scopy( value, y, -1 );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a first argument which is a read-only ndarray', function test( t ) {
```

Comment:
@aman-095 This is incorrect. The first argument is not mutated, so it can be read-only. This test passes because you chose to use `swap-factory`, but that tool is designed for different behavior.